diff --git a/node_modules/expo-media-library/ios/next/extensions/PHAssetExtensions.swift b/node_modules/expo-media-library/ios/next/extensions/PHAssetExtensions.swift
index c52d109..8ff9ea7 100644
--- a/node_modules/expo-media-library/ios/next/extensions/PHAssetExtensions.swift
+++ b/node_modules/expo-media-library/ios/next/extensions/PHAssetExtensions.swift
@@ -5,6 +5,7 @@ extension PHAsset {
     options: PHContentEditingInputRequestOptions = PHContentEditingInputRequestOptions()
   ) async throws -> (input: PHContentEditingInput?, info: [AnyHashable: Any]?) {
     return try await withCheckedThrowingContinuation { continuation in
+      options.isNetworkAccessAllowed = true
       self.requestContentEditingInput(with: options) { contentInput, info in
         continuation.resume(returning: (input: contentInput, info: info))
       }
